### PR TITLE
GPIOx cleanup and improvement

### PR DESCRIPTION
**Pull request by larchuto**
- STM32WL family does not depend on gpio_v2_common directly anymore
- Make it clear that STM32G0 depends on the same file as STM32L0
- Single matching pattern for GPIOx.LCKR.LCK[0-15] and GPIOx.BRR.BR[0-15]
- Apply GPIOx.BRR patch to STM32G0 and STM32L0 and STM32L5

### CODE REVIEW

**Comment by burrbull** [comment]
Please, rebase your branch on master.

**Comment by larchuto** [comment]
New modifications :
- Separation of gpio peripheral styles into 3 dedicated folders: v1, v2 and v3. v1 correspond to the F1 style and v3 to the G0, L0 and L5 style (it is basically the same as v2 but ST changed some names).
- gpio_go_l0 and gpio_l5 now share some code. This as the effect of slightly modifying the GPIOx.MODER.MODE* fields (removing the "(reset state)" mention, which carries information already given in the resetValue field).

**Comment by larchuto** [comment]
It may be better to put the v1,v2,v3 stuff into a separate PR ?

**Comment by larchuto** [comment]
@adamgreig or @newAM, would you possibly have some spare time to review this ?

**Comment by larchuto** [comment]
I just rebased it on top of master and resolved conflicts.

**Review by newAM — DISMISSED** [review]
Thanks for the pull-request!

This looks good to me; I will wait for another person to review because there are a lot of changes here.

File: peripherals/gpio/gpio_wl.yaml [hunk 1]
```diff
@@ -1,22 +1,17 @@
...
+  - v2/common.yaml
 
 "GPIO[AB]":
   BRR:
-    "BR[0123456789]":
-      _write:
-        NoAction: [0, "No action on the corresponding ODx bit"]
-        Reset: [1, "Reset the ODx bit"]
-    "BR1[012345]":
+    "BR[0123456789],BR1[012345]":
```
Comment (burrbull) [inline peripherals/gpio/gpio_wl.yaml:8]:
```suggestion
    "BR[0-9],BR1[0-5]":
```

Comment (larchuto) [reply peripherals/gpio/gpio_wl.yaml]: I've done the requested modifications in [a1b7ff3](https://github.com/stm32-rs/stm32-rs/commit/a1b7ff3f444898f1ddad9348273cb7b9fc44b03f).

**Review by newAM — APPROVED** [review]
Checked the generated output, overall much less code; 10107 insertions(+), 16189 deletions(-); lots more variants alias each other.  That's really great because GPIOs take the longest to compile because they have the largest volume of code.

bors r+

**Pull request merged**